File: solutions/0682-baseball-game/solution.js
/**
 * @param {string[]} operations
 * @return {number}
 */
var calPoints = function(operations) {
    let res = [];
    let sum = 0;
    for(let i = 0;i<operations.length;i++){
        if(operations[i] === "+"){
            res.push(res[res.length - 1] + res[res.length-2]);
        }
        else if(operations[i] === "D"){
            res.push(res[res.length-1] * 2);
        }
        else if(operations[i] === "C"){
            res.pop();
        }
        else{
            res.push(Number(operations[i]));
        }

    }
    for(let i = 0;i<res.length;i++){
        sum += res[i];
    }
    return sum;
};

File: solutions/0682-baseball-game/solution.cpp
class Solution {
public:
    int calPoints(vector<string>& operations) {
        vector <int> res;
        for(string c : operations){
            if(c == "+"){
                int score = res[res.size()-1] + res[res.size()-2];
                res.push_back(score);
            }
            else if(c == "D"){
                int score = res[res.size()-1]*2;
                res.push_back(score);
            }
            else if(c == "C"){
                res.pop_back();
            }
            else{
                res.push_back(stoi(c));
            }
        }
        int sum = 0;
        for(int i : res){
            sum += i;
        }
        return sum;
    }
};
